### DISCUSSION TITLE
Enable HTTPS using caddy on a docker install of remotely

### DISCUSSION THREAD

**Discussion by pieter831121** [immense/Remotely#665] [General]
Hi.

I used the docker instance to install remotely with default settings 

Can anyone help me with getting caddy setup so that i can use the site with https or any other alternative.

Regards

Pieter

**Comment by MKbyte** [comment]
Is Apache2 installed on your system?

**Comment by Tux1991** [comment]
Hi,

i made nginx reverse Proxy in a container on another host. It works. By the way you can build your nginx in another container on the same host and let the containers communicate with each other.

tuxya

**Comment by ajshea** [comment]
Install Caddy, https://caddyserver.com/docs/install#debian-ubuntu-raspbian 

Then see https://caddyserver.com/docs/running#using-the-service  

Create a Caddyfile `/etc/caddy/Caddyfile`

edit it with nano or vi, and the only two lines you need: 
```
support.example.com {
  reverse_proxy 127.0.0.1:5000
}
```
Restart Caddy: `service caddy restart` 

If your DNS is resolving for that name and IP address, Caddy will automatically request LetsEncrypt certificates for that address. Remotely will pick up the address and configure the download agents on the fly. 

Also see notes at https://hub.docker.com/r/immybot/remotely

**Comment by daveymg-nz** [comment]
If you have docker setup then running Caddy as a docker image is easiest. Just make sure that you use the remotley container name in the caddy config and not localhost. Caddy also requires that it is contactable on port 443 for https and 80 for authorising the certificate request.
